app/src/main/java/com/sdp/movemeet/utility/ImageHandler.java

```diff
@@ -0,0 +1,71 @@
+package com.sdp.movemeet.utility;
+
+import android.net.Uri;
+import android.util.Log;
+import android.view.View;
+import android.widget.ProgressBar;
+
+import androidx.annotation.NonNull;
+
+import com.google.android.gms.tasks.OnFailureListener;
+import com.google.android.gms.tasks.OnSuccessListener;
+import com.google.android.gms.tasks.Task;
+import com.google.firebase.storage.UploadTask;
+import com.sdp.movemeet.backend.BackendManager;
+import com.sdp.movemeet.backend.firebase.storage.StorageImageManager;
+import com.sdp.movemeet.models.Image;
+import com.squareup.picasso.Picasso;
+
+public class ImageHandler {
+
+    private static final String TAG = "FirebaseInteraction";
+
+    private static BackendManager<Image> imageBackendManager;
+
+    public static void loadImage(Image image, ProgressBar progressBar) {
+        if (progressBar != null) {
+            progressBar.setVisibility(View.VISIBLE);
+        }
+        imageBackendManager = new StorageImageManager();
+        Task<Uri> document = (Task<Uri>) imageBackendManager.get(image.getDocumentPath());
+        document.addOnSuccessListener(new OnSuccessListener<Uri>() {
+            @Override
+            public void onSuccess(Uri uri) {
+                Log.d(TAG, "Image successfully fetched from Firebase Storage!");
+                image.setImageUri(uri);
+                Picasso.get().load(uri).into(image.getImageView());
```

Comment:
I wil have to change the dependency of Picasso to Glide, as it's the only way for me to make it work the cache.

Reply:
Ok ok, no problem, if you manage to make `ImageHandler` still works with Glide then! Personally I use Glide only for the temporary spinning wheel when we are sending an image via the chat.